Tests/tests_e2e/content/xsoar_saas/test_e2e_xsoar_saas.py

```diff
@@ -176,15 +183,27 @@ def test_slack_ask(request: SubRequest, xsoar_saas_client: XsoarSaasClient):
             playbook_path="Tests/tests_e2e/content/xsoar_saas/TestSlackAskE2E.yml",
             playbook_id=playbook_id_name,
             playbook_name=playbook_id_name
-        ), save_incident(xsoar_saas_client, playbook_id=playbook_id_name) as incident_response:
-            # make sure the playbook finished successfully
-            assert xsoar_saas_client.poll_playbook_state(
-                incident_response.id, expected_states=(InvestigationPlaybookState.COMPLETED,)
-            )
+        ):
+            for i in range(1, 6):
+                try:
+                    with save_incident(xsoar_saas_client, playbook_id=playbook_id_name) as incident_response:
+                        # make sure the playbook finished successfully
+                        assert xsoar_saas_client.poll_playbook_state(
+                            incident_response.id, expected_states=(InvestigationPlaybookState.COMPLETED,)
+                        )
 
-            context = xsoar_saas_client.get_investigation_context(incident_response.investigation_id)
-            # make sure the context is populated with thread id(s) from slack ask
-            assert context.get("Slack", {}).get("Thread"), f'thread IDs do not exist in context {context}'
+                        context = xsoar_saas_client.get_investigation_context(incident_response.investigation_id)
+                        # make sure the context is populated with thread id(s) from slack ask
+                        assert context.get("Slack", {}).get("Thread"), f'thread IDs do not exist in context {context}'
+                        break
+                except Exception as error:
+                    logging.error(f"test_slack_ask - time {i} - error\n{error}")
+                    # print incident raw response to get info about it in case of failures
+                    logging.debug(
+                        f'test_slack_ask incident metadata:\n{xsoar_saas_client.search_incidents(incident_response.id)}'
+                    )
+                    if i == 5:
```

Comment:
I think this if == 5 statement should be added to the EDL test as well

Reply:
nice catch, fixed it